### PR TITLE
Improve docs for scoped threads

### PR DESCRIPTION
**Pull request by ghost**
The olds docs were really dated and overly wordy, IMO.

### CODE REVIEW

File: src/thread.rs [hunk 1]
```diff
@@ -1,111 +1,37 @@
...
-//!
-//! 1. Iterate over some collection.
-//! 2. Spin up a thread to operate on each part of the collection.
-//! 3. Join all the threads.
-//!
-//! However, this code actually gives an error:
-//!
-//! ```text
-//! error: `array` does not live long enough
-//! for i in &array {
```
Comment (jeehoonkang) [inline src/thread.rs]: We're removing the (hypothetical) error message here.  I'm not strongly against this change, but I think this error message can help readers to understand the motivation behind scoped threads.  What do you think?

Comment (stjepang) [reply src/thread.rs]: I've just added a section that illustrates why we need scoped threads. Is it better now?

**Comment by stjepang** [comment]
Just a ping for review - do the docs look better now?

**Comment by jeehoonkang** [comment]
Sorry for inactivity.  I like the new doc.  Merging it now!

**Review by jeehoonkang — APPROVED** [review]

**Pull request merged**